pallets/automation-time/src/lib.rs

```diff
@@ -336,90 +323,99 @@ pub mod pallet {
 		/// Trigger tasks for the block time.
 		///
 		/// Complete as many tasks as possible given the maximum weight.
-		/// We first check to see if there are any tasks that weren't completed in the last slot.
-		/// If this is the last block in the slot then we move the tasks to overflow.
-		/// Return the weight that was used.
-		///
-		/// Until the TODO is completed we will be limited to two tasks per slot to ensure
-		/// they can all be completed.
-		///
-		/// TODO (ENG-157):
-		/// - Calculate weights for each task
+		/// TODO (ENG-157): calculate weights.
 		pub fn trigger_tasks(max_weight: Weight) -> Weight {
-			let mut weight_left: Weight = max_weight - RUN_TASK_OVERHEAD;
-			let (time_slot, last_block_in_slot) = Self::get_current_time_slot();
+			// need to calculate cost of all but the inner IF.
+			let mut weight_left: Weight = max_weight - 20_000;
 
-			weight_left = Self::run_overflow_tasks(weight_left);
+			// There is a chance we use more than our max_weight to update the task queue.
+			// This would occur if the system is not producting blocks for a very long time.
+			// Regardless of how long it takes we still need to update the task queue.
+			let update_weight = Self::update_task_queue();
 
-			if weight_left < MAX_LOOP_WEIGHT {
-				if last_block_in_slot {
-					if let Some(task_ids) = Self::get_scheduled_tasks(time_slot) {
-						Self::move_to_overflow(task_ids.into_inner());
-						<ScheduledTasks<T>>::remove(time_slot);
-					}
-				}
-				return max_weight - weight_left
+			// need to calculate the weight of running just 1 task below.
+			if weight_left < update_weight + 10_000 {
+				return update_weight
+			} else {
+				weight_left -= update_weight;
 			}
 
-			weight_left = Self::run_scheduled_tasks(time_slot, last_block_in_slot, weight_left);
+			let task_queue = Self::get_task_queue();
+
+			if task_queue.len() > 0 {
+				// calculate cost of all but the run_tasks fcn.
+				weight_left -= 10_000;
```

Comment:
assuming this is is a TODO for future, since this isn't quite the calculation of run_tasks fcn and invocation of run tasks is right below.

also what if weight left is less than the calculation of the task queue? how should we handle that error? or do we have to loop through it to figure out what's our stopping point? although i do see `tasks_left` below, so are we just leaving it to run_tasks to figure out what's remaining?

Reply:
`run_tasks` loops until there is no weight left or the queue is empty